warp/Network/Wai/Handler/Warp/Request.hs

```diff
@@ -50,11 +51,13 @@ import Network.Wai.Handler.Warp.Settings (
 
 ----------------------------------------------------------------
 
+-- | first request on this connection?
+data FirstRequest = FirstRequest | SubsequentRequest
+
 -- | Receiving a HTTP request from 'Connection' and parsing its header
 --   to create 'Request'.
 recvRequest
```

Comment:
Apparently, this  is re-exported from `Network.Wai.Handler.Warp.Internal`.  @kazu-yamamoto @Vlix what are the stability guarantees for that module?

Reply:
We can change `Internal` at anytime without major version up.